src/library_syscall.js

```diff
@@ -1061,7 +1061,7 @@ function wrapSyscallFunction(x, library, isWasi) {
     pre += 'try {\n';
     handler +=
     "} catch (e) {\n" +
-    "  if (typeof FS == 'undefined' || !(e instanceof FS.ErrnoError)) throw e;\n";
+    "  if (typeof FS == 'undefined' || !(e.name === 'ErrnoError')) throw e;\n";
```

Comment:
I think we maybe need to think a little more carefully about this.  I think maybe the arguments made in https://github.com/emscripten-core/emscripten/pull/16994#discussion_r960379197 may apply here.

If we do want to make this code work between 2 different emscripten instances I think we should have some testing of that mode, and also make sure that doesn't effect the code size for folks who don't need/want that.

Reply:
I agree with @sbc100 that a test would be good for this. Otherwise I think this makes sense to do for the reasons you said @iliakan - `instanceof` is just brittle in JS, unfortunately (due to iframes etc.).

Reply:
> I agree with @sbc100 that a test would be good for this. Otherwise I think this makes sense to do for the reasons you said @iliakan - `instanceof` is just brittle in JS, unfortunately (due to iframes etc.).

So you suggest to write a test that fails without my PR, but succeeds with it?

Reply:
I'm actually not sure about this patch at all.   If you are trying to import/use the FS object from another module why wouldn't you also import its `FS.ErrnoError`?   In that mode wouldn't you want to completely avoid declaring `FS` (or `FS.ErrnoError`) in the consumer module?

Reply:
i.e. if you want module to somehow share and FS then only one of them should declare the FS, right?

Reply:
I added the test that demonstrates the bug on the current HEAD, and it's fixed in my branch.

P.S. Thank you for fast consideration of the PR. I almost forgot what the issue was exactly, but easily managed to remember and write it out.